main.go

```diff
@@ -225,6 +231,41 @@ func main1() int {
 	return 0
 }
 
+func validateOptions() error {
```

Comment:
Name is too generic

Reply:
Also, if you want to do this in a separate function, you might as well implement flag.Value which will be self-explanatory and automatically validate as part of flag.Parse.

Reply:
Oh cool, didn't know about `flag.Value`